### DISCUSSION TITLE
No module named 'finn.core.modelwrapper' and No module named 'finn.transformation.general'

### DISCUSSION THREAD

**Discussion by Ajfil** [Xilinx/finn#788] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/91005367/228166636-a5dcf945-7ba3-4a3e-9ffd-85102142dbad.png)

![image](https://user-images.githubusercontent.com/91005367/228166269-bba97fc9-f5b3-49a2-9ee9-b3f506aa1633.png)
hi，Why don't I show this module here？

**Comment by sefaburakokcu** [comment, marked as answer]
Hi @Ajfil,

Thank you for your interest. In https://github.com/sefaburakokcu/finn-quantized-yolo, we used Finn v0.7 for transformations. However, transformation and modelwrapper modules are being exported from qonnx instead of finn in new Finn versions(v0.9).

You can replace finn by qonnx if you are using Finn v0.9.  For example: 
from qonnx.core.modelwrapper import ModelWrapper
from qonnx.transformation.infer_data_layouts import InferDataLayouts
from qonnx.transformation.infer_datatypes import InferDataTypes
from qonnx.transformation.infer_shapes import InferShapes
from qonnx.transformation.insert_topk import InsertTopK
from qonnx.transformation.lower_convs_to_matmul import LowerConvsToMatMul
from qonnx.transformation.merge_onnx_models import MergeONNXModels
from qonnx.transformation.remove import RemoveIdentityOps

**Reply by josphat552** [reply]
Yes that solves the issue .

**Reply by Ajfil** [reply]
Thank your help
